src/devices/Bmxx80/Bmxx80Base.cs

```diff
@@ -48,67 +52,71 @@ protected Bmxx80Base(byte deviceId, I2cDevice i2cDevice)
+        /// Gets or sets the temperature sampling.
         /// </summary>
-        /// <param name="sampling">The <see cref="Sampling"/> to set.</param>
-        public void SetTemperatureSampling(Sampling sampling)
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <see cref="Sampling"/> is set to an undefined mode.</exception>
+        public Sampling TemperatureSampling
         {
-            byte status = Read8BitsFromRegister(_controlRegister);
-            status = (byte)(status & 0b0001_1111);
-            status = (byte)(status | (byte)sampling << 5);
-            _i2cDevice.Write(new[] { _controlRegister, status });
+            get => _temperatureSampling;
+            set
+            {
+                byte status = Read8BitsFromRegister(_controlRegister);
+                status = (byte)(status & 0b0001_1111);
+                status = (byte)(status | (byte)value << 5);
+
+                Span<byte> command = stackalloc[] {_controlRegister, status};
+                _i2cDevice.Write(command);
+                _temperatureSampling = value;
+            }
         }
 
         /// <summary>
-        /// Get the current sample rate for pressure measurements
+        /// When called, the device is reset using the complete power-on-reset procedure.
+        /// The device will reset to the default configuration.
         /// </summary>
-        /// <returns>The current pressure <see cref="Sampling"/> rate.</returns>
-        public Sampling ReadPressureSampling()
+        public void Reset()
         {
-            byte status = Read8BitsFromRegister(_controlRegister);
-            status = (byte)((status & 0b0001_1100) >> 2);
+            const byte resetCommand = 0xB6;
+            Span<byte> command = stackalloc[] {(byte)Bmxx80Register.RESET, resetCommand};
+            _i2cDevice.Write(command);
 
-            return ByteToSampling(status);
+            SetDefaultConfiguration();
         }
 
         /// <summary>
-        /// Get the sample rate for temperature measurements.
+        /// Read the temperature.
         /// </summary>
-        /// <returns>The current temperature <see cref="Sampling"/> rate.</returns>
-        public Sampling ReadTemperatureSampling()
-        {
-            byte status = Read8BitsFromRegister(_controlRegister);
-            status = (byte)((status & 0b1110_0000) >> 5);
-
-            return ByteToSampling(status);
-        }
+        /// <returns>Calculated temperature.</returns>
+        public abstract bool TryReadTemperature(out Temperature temperature);
 
         /// <summary>
-        /// When called, the device is reset using the complete power-on-reset procedure.
+        /// Read the pressure.
         /// </summary>
-        public void Reset()
-        {
-            const byte resetCommand = 0xB6;
-            _i2cDevice.Write(new[] { (byte)Bmxx80Register.RESET, resetCommand });
-        }
+        /// <returns>Calculated pressure in Pa.</returns>
```

Comment:
NIT: doc is not correct. This actually returns a bool which I suppose means success, and has an out parameter which is the one that has the calculated pressure.

Reply:
Ah yes good catch